src/bluesearch/mining/eval.py

```diff
@@ -423,34 +486,40 @@ def ner_report(iob_true, iob_pred, mode="entity", etypes_map=None, return_dict=F
 
 
 def ner_errors(
-    iob_true, iob_pred, tokens, mode="entity", etypes_map=None, return_dict=False
-):
+    iob_true: pd.Series,
+    iob_pred: pd.Series,
+    tokens: pd.Series,
+    mode: Optional[str] = "entity",
```

Comment:
I might be wrong here, but I think `Optional[str]` is equivalent to `Union[str, None]`. IMO it means that the user can pass a `None` and the code will handle it accordingly (it does not mean "this is a keyword argument that has a default"). However, the code cannot handle `mode=None` and would raise an exception.

See https://mypy.readthedocs.io/en/stable/getting_started.html#the-typing-module for references.

I feel like the meaning is different from the `numpydoc`'s `optional`.

Reply:
Very interesting, I totally missed this and I indeed thought it was like `numpydoc`'s `optional!

I think this also applies to this line (even if it's not from this PR):
https://github.com/BlueBrain/Search/blob/b9c886966ca4d893b41457a17262e198e3ba7f03/src/bluesearch/embedding_models.py#L303

Fixed all wrong usages of `Optiona[...]` in 1ecd77b.